scripts/config

```diff
@@ -0,0 +1,257 @@
+
+#=================================================
+# SHOW_CONFIG FUNCTION FOR 'SHOW' COMMAND
+#=================================================
+
+show_config() {
+	# here you are supposed to read some config file/database/other then print the values
+	# echo "YNH_CONFIG_${PANEL_ID}_${SECTION_ID}_${OPTION_ID}=value"
+
+	echo "YNH_CONFIG_MAIN_EXPORT_EXPORT=$export"
+
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_NOCOLORS=$pad_config_nocolors"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_SHOWLINENUMBERS=$pad_config_showlinenumbers"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_CHATANDUSERS=$pad_config_chatandusers"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_ALWAYSSHOWCHAT=$pad_config_alwaysshowchat"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_SHOW_MARKDOWN=$pad_config_show_markdown"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_PAGEVIEW=$pad_config_pageview"
+
+	echo "YNH_CONFIG_MAIN_MYPADS_CONFIGURATION_MYPADS=$mypads"
+	echo "YNH_CONFIG_MAIN_MYPADS_CONFIGURATION_USELDAP=$useldap"
+
+	echo "YNH_CONFIG_MAIN_IS_PUBLIC_IS_PUBLIC=$is_public"
+}
+
+#=================================================
+# MODIFY THE CONFIGURATION
+#=================================================
+
+apply_config() {
+	restart_etherpad=0
+
+	# Change configuration if needed
+	# padOptions noColors
+	if [ "$pad_config_nocolors" != "$old_pad_config_nocolors" ]
+	then
+		ynh_replace_string "\(\"noColors\" *: \).*," "\1$pad_config_nocolors," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions showLineNumbers
+	if [ "$pad_config_showlinenumbers" != "$old_pad_config_showlinenumbers" ]
+	then
+		ynh_replace_string "\(\"showLineNumbers\" *: \).*," "\1$pad_config_showlinenumbers," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions chatAndUsers
+	if [ "$pad_config_chatandusers" != "$old_pad_config_chatandusers" ]
+	then
+		ynh_replace_string "\(\"chatAndUsers\" *: \).*," "\1$pad_config_chatandusers," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions alwaysShowChat
+	if [ "$pad_config_alwaysshowchat" != "$old_pad_config_alwaysshowchat" ]
+	then
+		ynh_replace_string "\(\"alwaysShowChat\" *: \).*," "\1$pad_config_alwaysshowchat," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Plugin option ep_markdown_default
+	if [ "$pad_config_show_markdown" != "$old_pad_config_show_markdown" ]
+	then
+		ynh_replace_string "\(\"ep_markdown_default\" *: \).*," "\1$pad_config_show_markdown," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Plugin option ep_page_view_default
+	if [ "$pad_config_pageview" != "$old_pad_config_pageview" ]
+	then
+		ynh_replace_string "\(\"ep_page_view_default\" *: \).*," "\1$pad_config_pageview," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Export
+	if [ "$export" != "$old_export" ]
+	then
+		if [ "$export" = "abiword" ]
+		then
+			# if abiword isn't installed, call the action add_remove_abiword.
+			if ! which abiword > /dev/null
```

Comment:
Are we really sure that if `which abiword` return true, abiword is installed as etherpad package dependances ?

I purpose to use something more like `apt-cache show etherpad_ynh-deps | egrep "Depends: .*abiword"`

Or just with `dpkg -l | grep ...`

Reply:
abiword can be a dependence of etherpad, https://github.com/YunoHost-Apps/etherpad_mypads_ynh/blob/master/scripts/install#L89-L93
It's depend of the install, or of this action script.

I've tested it, `which abiword` will return false if abiword isn't installed.

Reply:
If the user has typed `apt-get install abiword`, either abiword is a dependence of etherpad or not doesn't matter. Removing etherpad won't remove abiword, since abiword will then be installed 'manually' for apt.
Abiword has to be a dependence of etherpad if it's installed by the ynh package. Otherwise, it doesn't matter.

What really matters here is to know if abiword is available or not. And if it's not, to install it.